tests/test_server.py

```diff
@@ -91,6 +100,29 @@ def test_remote_butler(self):
         universe = self.butler.dimensions
         self.assertEqual(universe.namespace, "daf_butler")
 
+    def test_instantiate_via_butler_http_search(self):
+        """Ensure that the primary Butler constructor's automatic search logic
+        correctly locates and reads the configuration file and ends up with a
+        RemoteButler pointing to the correct URL
+        """
+
+        # This is kind of a fragile test.  Butler's search logic does a lot of
+        # manipulations involving creating new ResourcePaths, and ResourcePath
+        # doesn't use httpx so we can't easily inject the TestClient in there.
```

Comment:
If we decide that we want to do that we can consider changing ResourcePath to use `httpx` rather than `requests`.